src/StatsdClient/BufferBuilder.cs

```diff
@@ -0,0 +1,45 @@
+using System.Text;
+
+namespace StatsdClient
+{
+    /// <summary>
+    /// Append string values to a fixed size bytes buffer.
+    /// </summary>
+    class BufferBuilder
+    {
+        readonly IBufferBuilderHandler _handler;
+        readonly byte[] _buffer;
+
+        public BufferBuilder(IBufferBuilderHandler handler, int bufferCapacity)
+        {
+            _buffer = new byte[bufferCapacity];
+            _handler = handler;
+        }
+
+        public int Length { get; private set; }
+        public int Capacity { get { return _buffer.Length; } }
+
+        public bool Add(string value)
+        {
+            var byteCount = Encoding.UTF8.GetByteCount(value);
+            if (byteCount > _buffer.Length)
```

Comment:
Using `Capacity` instead of `_buffer.Length` might read more cleanly?

Reply:
Done.